tests/kserve_models_web_application_test.sh

```diff
@@ -0,0 +1,169 @@
+#!/bin/bash
+set -euxo pipefail
+
+# KServe Models Web Application Test Script
+# Tests the models-web-application API functionality with kubectl-deployed InferenceServices
+
+KF_PROFILE=${1:-kubeflow-user-example-com}
+TOKEN="$(kubectl -n $KF_PROFILE create token default-editor)"
+BASE_URL="localhost:8080/kserve-endpoints"
+
+echo "=========================================="
+echo "KServe Models Web Application Test"
+echo "Profile: ${KF_PROFILE}"
+echo "=========================================="
+
+# Pre-Test Setup: Configure models-web-application to disable authentication for testing
+kubectl patch configmap kserve-models-web-app-config -n kubeflow \
+  --type merge \
+  -p '{"data":{"APP_DISABLE_AUTH":"True"}}'
+
+kubectl rollout restart deployment kserve-models-web-app -n kubeflow
+kubectl rollout status deployment kserve-models-web-app -n kubeflow --timeout=120s
+
+# Pre-Test Setup: Create RBAC for default-editor to access InferenceServices
+echo ""
+echo "Step 2: Setting up RBAC permissions for InferenceService access..."
+cat <<EOF | kubectl apply -f -
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: inferenceservice-editor
+  namespace: ${KF_PROFILE}
+rules:
+- apiGroups:
+  - serving.kserve.io
+  resources:
+  - inferenceservices
+  verbs:
+  - get
+  - list
+  - watch
+  - create
+  - delete
+  - deletecollection
+  - patch
+  - update
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: default-editor-inferenceservice-access
+  namespace: ${KF_PROFILE}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: inferenceservice-editor
+subjects:
+- kind: ServiceAccount
+  name: default-editor
+  namespace: ${KF_PROFILE}
+EOF
```

Comment:
Why do you add this? the default editor already should have the permissions to modify kserve inferenceservices